File: app/src/main/java/com/br/rafael/TesteAndroidv2/ui/loginScreen/LoginConfigurator.java
package com.br.rafael.TesteAndroidv2.ui.loginScreen;

import java.lang.ref.WeakReference;

public enum LoginConfigurator {

    INSTANCE;

    public void configure(LoginActivity activity){

        LoginRouter router = new LoginRouter();
        router.activity = new WeakReference<>(activity);

        LoginPresenter presenter = new LoginPresenter();
        presenter.output = new WeakReference<LoginActivityInput>(activity);

        LoginInteractor interactor = new LoginInteractor();
        interactor.output = presenter;

        if (activity.output == null){
            activity.output = interactor;
        }
        if (activity.router == null){
            activity.router = router;
        }

    }
}

File: SantanderApp/app/src/main/java/com/qintess/santanderapp/ui/view/loginScreen/LoginConfigurator.kt
package com.qintess.santanderapp.ui.view.loginScreen

import java.lang.ref.WeakReference

enum class LoginConfigurator {
    INSTANCE;

    fun configure(activity: LoginActivity) {
        val presenter = LoginPresenter()
        presenter.output = WeakReference(activity)

        val interactor = LoginInteractor()
        interactor.output = presenter

        if (activity.output == null) {
            activity.output = interactor
        }
    }
}